### DISCUSSION TITLE
Plugins in directory mode

### DISCUSSION THREAD

**Discussion by mobab-th** [simonw/datasette#1317] [Q&A] [answered]
Hello,

how can i install / use Plugins like datsette-media in directory mode?

**Comment by brandonrobertz** [comment, marked as answer]
The `plugins` directory in directory mode only supports single-file plugins. If the plugin you're trying to install isn't a single file, you need to install your plugin using one of the normal methods: pip install (e.g., `pip install datasette-media`) or datasette install (e.g., `datasette install datasette-media`)

Try and think of the "plugins" in directory mode less as a full blown library or module, but as python scripts that happen to use one of Datasette's [plugin hooks](https://docs.datasette.io/en/stable/plugin_hooks.html).

**Reply by mobab-th** [reply]
Thanks
